sdk/resourcemanager/Azure.ResourceManager/src/autorest.md

```diff
@@ -52,56 +52,90 @@ rename-rules:
   URI: Uri
 
 directive:
-  - remove-model: "AzureEntityResource"
-  - remove-model: "ProxyResource"
-  - remove-model: "ResourceModelWithAllowedPropertySet"
-  - remove-model: "Identity"
-  - remove-model: "Operation"
-  - remove-model: "OperationListResult"
-  - remove-model: "OperationStatusResult"
-  - remove-model: "locationData"
-  - remove-model: "CheckNameAvailabilityRequest"
-  - remove-model: "CheckNameAvailabilityResponse"
-  - remove-model: "ErrorResponse"
-  - remove-model: "ErrorDetail"
-  - remove-model: "ErrorAdditionalInfo"
   - from: types.json
-    where: $.definitions['Resource']
+    where: $.definitions.Resource
     transform: >
-      $["x-ms-mgmt-referenceType"] = true
+      $["x-ms-mgmt-referenceType"] = true;
+      $["x-ms-mgmt-propertyReferenceType"] = true;
+      $["x-namespace"] = "Azure.ResourceManager.Models";
+      $["x-accessibility"] = "public";
+      $["x-csharp-formats"] = "json";
+      $["x-csharp-usage"] = "model,input,output";
   - from: types.json
-    where: $.definitions['TrackedResource']
+    where: $.definitions.TrackedResource
     transform: >
-      $["x-ms-mgmt-referenceType"] = true
+      $["x-ms-mgmt-referenceType"] = true;
+      $["x-ms-mgmt-propertyReferenceType"] = true;
+      $["x-namespace"] = "Azure.ResourceManager.Models";
+      $["x-accessibility"] = "public";
+      $["x-csharp-formats"] = "json";
+      $["x-csharp-usage"] = "model,input,output";
   - from: types.json
-    where: $.definitions.*
+    where: $.definitions.Plan
     transform: >
       $["x-ms-mgmt-propertyReferenceType"] = true;
       $["x-namespace"] = "Azure.ResourceManager.Models";
       $["x-accessibility"] = "public";
       $["x-csharp-formats"] = "json";
       $["x-csharp-usage"] = "model,input,output";
   - from: types.json
-    where: $.definitions.*.properties[?(@.enum)]
+    where: $.definitions.Sku
     transform: >
+      $["x-ms-mgmt-propertyReferenceType"] = true;
       $["x-namespace"] = "Azure.ResourceManager.Models";
       $["x-accessibility"] = "public";
-# Workaround for the issue that SystemData lost readonly attribute: https://github.com/Azure/autorest/issues/4269
+      $["x-csharp-formats"] = "json";
+      $["x-csharp-usage"] = "model,input,output";
+  - from: types.json
+    where: $.definitions.systemData
+    transform: >
+      $["x-ms-mgmt-propertyReferenceType"] = true;
+      $["x-namespace"] = "Azure.ResourceManager.Models";
+      $["x-accessibility"] = "public";
+      $["x-csharp-formats"] = "json";
+      $["x-csharp-usage"] = "model,input,output";
   - from: types.json
     where: $.definitions.systemData.properties.*
     transform: >
-      $["readOnly"] = true
+      $["readOnly"] = true;
```

Comment:
I think we can update the `x-csharp-usage` to be `model,output`, then we can remove the below workaround to set each property in `systemData` as readOnly.

Reply:
I will have a try

Reply:
well, it works, but it also changes a constructor from public to internal, which is a breaking change. I would like to keep the previous implementation...